src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -5302,6 +5373,29 @@ def update_api_server_access_profile(self, mc: ManagedCluster) -> ManagedCluster
         mc.api_server_access_profile = profile_holder
         return mc
 
+    def update_windows_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update windows profile for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if not mc.windows_profile:
+            raise UnknownError(
+                "Encounter an unexpected error while getting windows profile from the cluster in the process of update."
+            )
+
+        enable_ahub = self.context.get_enable_ahub()
+        disable_ahub = self.context.get_disable_ahub()
+        windows_admin_password = self.context.get_windows_admin_password()
+        if enable_ahub:
```

Comment:
Could `enable_ahub` be `none`? We should not update ahub if neither --enable-ahub and --disable-ahub is used.

Reply:
No, `enable_ahub` could not be `None` since its default value is `False` defined [here](https://github.com/FumingZhang/azure-cli/blob/fuming/acs-update-np8/src/azure-cli/azure/cli/command_modules/acs/custom.py#L2363-L2364) and the user's input will only assign a value other than `None`.

Reply:
We only set license_type when either is true. You do not change that. LGTM

Reply:
By the way, I suppose `--enable-ahub` and `--disable-ahub` should be boolean flags instead of normal options which would take an argument? If so, I would like to add `action='store_true'` to the following places to express this clearly (by default, the option will be parsed as an ordinary parameter, and now it is implicitly inferred by the cli through the default value of the parameter in the function of the boolean type). 

- [create: --enable-ahub](https://github.com/FumingZhang/azure-cli/blob/fuming/acs-update-np8/src/azure-cli/azure/cli/command_modules/acs/_params.py#L262)
- [update: --enable-ahub](https://github.com/FumingZhang/azure-cli/blob/fuming/acs-update-np8/src/azure-cli/azure/cli/command_modules/acs/_params.py#L318)
- [update: --disable-ahub](https://github.com/FumingZhang/azure-cli/blob/fuming/acs-update-np8/src/azure-cli/azure/cli/command_modules/acs/_params.py#L319)

Reply:
They are boolean flags. Please do it.

Reply:
Changed in commit [#7b3cb27](https://github.com/Azure/azure-cli/pull/20145/commits/7b3cb2707dbc1b757c771d2d004ef45c7928b73b).